src/test/java/org/stellar/sdk/Sep10ChallengeTest.java

```diff
@@ -143,6 +143,34 @@ public void testReadChallengeTransactionInvalidNotSignedByServer() throws IOExce
     }
   }
 
+  @Test
+  public void testReadChallengeTransactionInvalidCorrupted() throws InvalidSep10ChallengeException, IOException {
+    KeyPair server = KeyPair.random();
+    KeyPair client = KeyPair.random();
+    Network network = Network.TESTNET;
+
+    long now = System.currentTimeMillis() / 1000L;
+    long end = now + 300;
+    TimeBounds timeBounds = new TimeBounds(now, end);
+
+    String challenge = Sep10Challenge.newChallenge(
+            server,
+            network,
+            client.getAccountId(),
+            "Stellar Test",
+            timeBounds
+    );
+
+    challenge = challenge.replace("A", "B");
+
+    try {
+      Sep10Challenge.readChallengeTransaction(challenge, server.getAccountId(), Network.TESTNET);
+      fail();
+    } catch (RuntimeException ignored) {
+      // TODO: I think we should add an exception signature about the failure to read XDR on Transaction.fromEnvelopeXdr.
+    }
```

Comment:
I think this improvement could happen in a separate PR. In this test i think we just need to assert that an exception occurs if there's nothing about the exception we can assert on.

Reply:
Actually this is already done here, please see line 168. If `Sep10Challenge.readChallengeTransaction` does not throw an exception, then the test will fail and `fail() function` will throw an AssertionError, so I don't think it needs to be modified here, we can keep this `TODO` as a reminder.

Reply:
Oh! Of cos! Thank you, and sorry I missed that.